Doc/library/functools.rst

```diff
@@ -512,6 +518,167 @@ The :mod:`functools` module defines the following functions:
+      will be the union of all dependencies passed in.
+
+      It is possible to add a node with no dependencies (*predecessors* is not
+      provided) as well as provide a dependency twice. If a node that has not been
+      provided before is included among *predecessors* it will be automatically added
+      to the graph with no predecessors of its own.
+
+      Raises :exc:`ValueError` if called after :meth:`~TopologicalSorter.prepare`.
+
+   .. method:: prepare()
+
+      Mark the graph as finished and check for cycles in the graph. If any cycle is
+      detected, :exc:`CycleError` will be raised, but
+      :meth:`~TopologicalSorter.get_ready` can still be used to obtain as many nodes
+      as possible until cycles block more progress. After a call to this function,
+      the graph cannot be modified and therefore no more nodes can be added using
+      :meth:`~TopologicalSorter.add`.
+
+   .. method:: is_active()
+
+      Returns ``True`` if more progress can be made and ``False`` otherwise. Progress
+      can be made if cycles do not block the resolution and either there are still
+      nodes ready that haven't yet been returned by
+      :meth:`TopologicalSorter.get_ready` or the number of nodes marked
+      :meth:`TopologicalSorter.done` is less than the number that have been returned
+      by :meth:`TopologicalSorter.get_ready`.
+
+      The :meth:`~TopologicalSorter.__bool__` method of this class defers to this
+      function, so instead of::
+
+          if ts.is_active():
+              ...
+
+      if possible to simply do::
+
+          if ts:
+              ...
+
+      Raises :exc:`ValueError` if called without calling :meth:`~TopologicalSorter.prepare`
+      previously.
+
+   .. method:: done(*nodes)
+
+      Marks a set of nodes returned by :meth:`TopologicalSorter.get_ready` as
+      processed, unblocking any successor of each node in *nodes* for being returned
+      in the future by a call to :meth:`TopologicalSorter.get_ready`.
+
+      Raises :exc:`ValueError` if any node in *nodes* has already been marked as
+      processed by a previous call to this method or if a node was not added to the
+      graph by using :meth:`TopologicalSorter.add`, if called without calling
+      :meth:`~TopologicalSorter.prepare` or if node has not yet been returned by
+      :meth:`~TopologicalSorter.get_ready`.
+
+   .. method:: get_ready()
+
+      Returns a ``tuple`` with all the nodes that are ready. Initially it returns all
+      nodes with no predecessors and once those are marked as processed by calling
+      :meth:`TopologicalSorter.done`, further calls will return all new nodes that
+      have all their predecessors already processed until no more progress can be
+      made.
+
+      Raises :exc:`ValueError` if called without calling
+      :meth:`~TopologicalSorter.prepare` previously.
+
+   .. method:: static_order()
+
+      Returns an iterable of nodes in a stable topological order. Using this method
```

Comment:
It's unclear what "stable" is intended to mean in this context.  Perhaps just remove the word?